docs/_dapp/mobile.md

```diff
@@ -1,180 +1,368 @@
+   to sign many transactions or the application would need to do so without requiring user
+   input. The crucial aspect to remember is that **anyone with access to the account
+   manager can sign transactions while a particular account is unlocked** (e.g. device
+   left unattended; application running untrusted code).
+
+
+### Signing on Android (Java)
+
+Assuming an instance of a `KeyStore` called `ks` exists, a new account to sign transactions 
+can be created using its `newAccount` method. For this demonstation a hard-coded 
+example transaction is created to sign:
+
+```java
+// Create a new account to sign transactions with
+Account signer = ks.newAccount("Signer password");
+Transaction tx = new Transaction(
+    1, new Address("0x0000000000000000000000000000000000000000"),
+    new BigInt(0), new BigInt(0), new BigInt(1), null); // Random empty transaction
+BigInt chain = new BigInt(1); // Chain identifier of the main net
+```
 
-#### Custom dependency
+The transaction `tx` can be signed using the authorization mechanisms described above:
 
-If you prefer not to depend on CocoaPods; or would like to access an older develop build
-not available any more as an online dependency, you can download any bundle directly from
-[our website](https://geth.ethereum.org/downloads/) and insert it into your project in
-XCode via `Project Settings -> Build Phases -> Link Binary With Libraries`.
+```java
+// Sign a transaction with a single authorization
+Transaction signed = ks.signTxPassphrase(signer, "Signer password", tx, chain);
 
-Do not forget to extract the framework from the compressed `.tar.gz` archive. You can do
-that either using a GUI tool or from the command line via (replace the archive with your
-downloaded file):
+// Sign a transaction with multiple manually cancelled authorizations
+ks.unlock(signer, "Signer password");
+signed = ks.signTx(signer, tx, chain);
+ks.lock(signer.getAddress());
 
+// Sign a transaction with multiple automatically cancelled authorizations
+ks.timedUnlock(signer, "Signer password", 1000000000);
+signed = ks.signTx(signer, tx, chain);
 ```
-tar -zxvf geth-ios-all-1.5.3-unstable-e05d35e6.tar.gz
+
+### Signing on iOS (Swift 3)
+
+Assuming an instance of a `GethKeyStore` called `ks` exists, a new account 
+can be created to sign transactions with its `newAccount` method. For 
+this demonstation a hard-coded example transaction is created to sign:
+
+```swift
+// Create a new account to sign transactions with
+var error: NSError?
+let signer = try! ks?.newAccount("Signer password")
+
+let to    = GethNewAddressFromHex("0x0000000000000000000000000000000000000000", &error)
+let tx    = GethNewTransaction(1, to, GethNewBigInt(0), GethNewBigInt(0), GethNewBigInt(0), nil) // Random empty transaction
+let chain = GethNewBigInt(1) // Chain identifier of the main net
 ```
 
-#### Manual builds
+*Although Swift usually rewrites `NSError` returns to throws, this particular
```

Comment:
can anyone confirm whether this note about `NSError` is still true?

Reply:
I don't know -- possibly @karalabe have some experience with IOS development, not sure who does